### ISSUE TITLE
streaming registerCQ miss records to insert when changing duration seconds and slide seconds

### ISSUE THREAD

**Issue by niko2014** [SnappyDataInc/snappydata#862]
# Env:
snappydata-0.9
-J-Dspark.streaming.backpressure.enabled=true
1 locator, 3 dataserver, 1 leader
stream Table created with `using kafka_stream`
Kafka version : `0.10.1.0` 

# code

I read from streamTableA `(duration 6 seconds, slide 6 seconds)` and write to columnTableA.

```
        val dStream = snsc.registerCQ("select * from streamTableA window (duration 6 seconds, slide 6 seconds)");
        import snsc.snappySession.sqlContext.implicits._
        dStream.foreachDataFrame(df => {
            df.as[SnappyImageRow].write.insertInto("columnTableA")
        })
```

# before streaming started : 

There are `1000000 * 3 = 3,000,000` records to consume (3 partition , each with 1000000 records )

kafka offset : 
```
[2017-09-19 11:02:48,916] WARN WARNING: ConsumerOffsetChecker is deprecated and will be dropped in releases following 0.9.0. Use ConsumerGroupCommand instead. (kafka.tools.ConsumerOffsetChecker$)
Group           Topic                          Pid Offset          logSize         Lag             Owner
snappyStreaming01 stored_capture_features_01     0   220505899       221505899       1000000         none
snappyStreaming01 stored_capture_features_01     1   219352291       220352291       1000000         none
snappyStreaming01 stored_capture_features_01     2   219307305       220307305       1000000         none
```

columnTableA row count: `25,034,770`

# streaming ended

kafka offset : 
```
[2017-09-19 11:24:25,679] WARN WARNING: ConsumerOffsetChecker is deprecated and will be dropped in releases following 0.9.0. Use ConsumerGroupCommand instead. (kafka.tools.ConsumerOffsetChecker$)
Group           Topic                          Pid Offset          logSize         Lag             Owner
snappyStreaming01 stored_capture_features_01     0   221505899       221505899       0               snappyStreaming01_node78-1505791154261-790ae794-0
snappyStreaming01 stored_capture_features_01     1   220352291       220352291       0               snappyStreaming01_node78-1505791154261-790ae794-0
snappyStreaming01 stored_capture_features_01     2   220307305       220307305       0               snappyStreaming01_node78-1505791154261-790ae794-0
```

columnTableA row count: `25,722,888`

# consumed records != inserted row count

columnTableA row count change :
25,722,888 - 25,034,770 = 688,118

688,118 != 3,000,000
There are only 688,118 records inserted in columnTableA . 

# duration 1 seconds and slide 1 seconds

When I changed `duration 6 seconds and slide 6 seconds` back to `duration 1 seconds and slide 1 seconds`, streaming worked `fine` ( 3,000,000 consumed, 3,000,000 inserted to columnTableA ).

Is this a feature or bug ?

**Comment by ymahajan** [comment]
@niko2014 We will try to reproduce this in-house. Is this something observed only when backpressure is enabled? With or without direct kafka? Which version of kafka are you using?

Also one quick change you could try to isolate the issue is to create two separate column tables - one for a CQ with 1 sec window and other for 6 second window and they will ingest to the respective column tables, and verify if there is a row mismatch in that case as well.

**Comment by niko2014** [comment]
@ymahajan  
1. dashboard UI 's `streaming` tab said : `Running batches of 1 second for 2 minutes 57 seconds since 2017/09/19 11:19:10 (174 completed batches, 3000000 records)`, jobs seems all successful.
2. I create stream Table with `using kafka_stream`.
3. Kafka version is `0.10.1.0` .
(Sorry. Forgot to post informations about Kafka. I have updated the issue now.)

**Comment by ymahajan** [comment]
@niko2014 Thanks for reporting, this issue has been fixed in the product now, the fix will be available in 1.0.1 release. closing this one.

**Comment by sumwale** [comment]
@ymahajan Can you mention the JIRA/PRs here?

**Comment by sumwale** [comment]
The PR having the fix to this seems to be #882 . @ymahajan is that correct?
